source/posix/thread.c

```diff
@@ -160,7 +160,7 @@ int aws_thread_launch(
  * NUMA or not is setup in interleave mode.
  * Thread afinity is also not supported on Android systems, and honestly, if you're running android on a NUMA
  * configuration, you've got bigger problems. */
-#if !defined(__MACH__) && !defined(__ANDROID__)
+#if !defined(__MACH__) && !defined(__ANDROID__) && defined(HAVE_PTHREAD_ATTR_SETAFFINITY_NP)
```

Comment:
This code is actually vital if you want to hit full NIC entitlement on numa architectures. Are there any ways we can check for the alternative variants and use those where we can? I suspect a lot of Rust and C++ users link against musl and wouldn't like to see a massive performance reduction because of it.

Reply:
Do you mean like falling back to `pthread_setaffinity_np`? I think that extension is supported on musl.

Reply:
Yes, that would work.

Reply:
In fact, that might also work for BSD variants.

Reply:
Okay, the patch is a bit more complicated with this fallback mechanism, but it's definitely valuable to add support for musl and maybe even BSD. I don't have a BSD setup to test on but maybe I can try it out in a VM later this week. Would appreciate if you could take a look at the new revision I pushed when you have the time.

Reply:
this looks a good deal better. Thanks! now we're going to have to work around github actions and move this to a branch so CI will run. I'll link it here.